e2e/e2etests/test_solana_deposit_call.go

```diff
@@ -0,0 +1,53 @@
+package e2etests
+
+import (
+	"math/big"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/e2e/utils"
+	testcontract "github.com/zeta-chain/zetacore/testutil/contracts"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// TestSolanaDepositAndCall tests deposit of lamports calling a example contract
+func TestSolanaDepositAndCall(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 1)
+
+	// parse deposit amount (in lamports)
+	// #nosec G115 e2e - always in range
+	depositAmount := big.NewInt(int64(parseInt(r, args[0])))
+
+	// deploy an example contract in ZEVM
+	contractAddr, _, contract, err := testcontract.DeployExample(r.ZEVMAuth, r.ZEVMClient)
```

Comment:
Not a block er for this one but I would like to keep a single contract for all test https://github.com/zeta-chain/node/issues/2655

It would be deployed outside of the test and address given in config, so you don't need to deploy everytime on live network

https://github.com/zeta-chain/node/issues/2655

Can we add a TODO comment here to remove this line?

Reply:
same commit https://github.com/zeta-chain/node/pull/2726/commits/92d67f5897576fc119b93838893b0aaf9a43ec08